tico/experimental/quantization/passes/insert_quantize_on_dtype_mismatch.py

```diff
@@ -271,6 +272,39 @@ def _insert_quantize_op_after(node):
                 elif qparam_dtype(inp) == "uint8" and qparam_dtype(node) == "int16":
                     quantize = _insert_quantize_op_after(node)
 
+                    quantize.meta[QPARAM_KEY] = copy.deepcopy(node.meta[QPARAM_KEY])
+                    node.meta[QPARAM_KEY] = _i16_to_u8(node.meta[QPARAM_KEY])
+                    logger.debug(
+                        f"quantize_per_tensor.default is inserted after {node.name}."
+                    )
+                else:
+                    raise NotYetSupportedError("Unsupported dtype")
+            elif node.target == torch.ops.aten.reshape.default:
+                reshape_args = ReshapeArgs(*node.args, **node.kwargs)
+                inp = reshape_args.input
+
+                if QPARAM_KEY not in inp.meta:
+                    continue
+
+                if QPARAM_KEY not in node.meta:
+                    continue
+
+                if qparam_dtype(inp) == qparam_dtype(node):
+                    continue
+
+                if qparam_dtype(inp) == "int16" and qparam_dtype(node) == "uint8":
+                    # A new Quantize Op (s16 to u8) is inserted before (not after)
+                    # reshape Op to reduce tensor size ealier
```

Comment:
I am new to this parts of codes, so could you please explain why int16->uint8 requires `insert quantize op before` and uint8->int16 requires `insert quantize op after` functions?

Reply:
Let me explain the rationale. For int16->uint8,

1. Insert Quantize before Reshape.

```
Predecessor (int16)-> Quantize (uint8) -> Reshape (uint8) -> ...
```

2. Insert Quantize after Reshape.

```
Predecessor (int16)-> Reshape (int16) -> Quantize (uint8) -> ...
```

Comparing 1) and 2), the difference is that Reshape operation is conducted in uint8 or int16.  I decided to go with 1), which does Reshape in uint8, for faster execution of Reshape. Note that Reshape Op does not change the value, so its dytpe does not affect accuracy.

uint8->int16 case is the opposite of the above example.

Reply:
@jinevening Thank you for the clear explanation :-D!!